server/app/query/view/[id]/page.tsx

```diff
@@ -86,18 +89,90 @@ export default function QueryPage({ params }: { params: { id: string } }) {
+      <div className="flex justify-between items-center">
+        <div>
+          <h2 className="text-2xl font-bold leading-7 text-gray-900 dark:text-gray-100 sm:truncate sm:text-3xl sm:tracking-tight">
+            Query Details
+          </h2>
+        </div>
+        <div>
+          <button
+            onClick={() => kill(IPARemoteServers)}
+            type="button"
+            className="ml-3 rounded-md bg-red-600 px-3 py-2 text-sm font-semibold text-white shadow-sm hover:bg-red-700 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-red-600"
+          >
+            Kill Query
+          </button>
+        </div>
+      </div>
 
-        <button
-          onClick={() => kill(IPARemoteServers)}
-          type="button"
-          className="ml-3 rounded-md bg-red-600 px-3 py-2 text-sm font-semibold text-white shadow-sm hover:bg-red-700 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-red-600"
-        >
-          Kill Query
-        </button>
+      <div className="mt-6 border-t border-b border-gray-300">
+        <dl className="divide-y divide-gray-200">
+          <div className="px-4 py-2 sm:grid sm:grid-cols-3 sm:gap-4 sm:px-0">
+            <dt className="text-sm font-medium leading-6 text-gray-900">
+              Display name:
+            </dt>
+            <dd className="mt-1 text-sm leading-6 text-gray-700 sm:col-span-2 sm:mt-0">
+              {params.id}
+            </dd>
+          </div>
+
+          <div className="px-4 py-2 sm:grid sm:grid-cols-3 sm:gap-4 sm:px-0">
+            <dt className="text-sm font-medium leading-6 text-gray-900">
+              UUID:
+            </dt>
+            <dd className="mt-1 text-sm leading-6 text-gray-700 sm:col-span-2 sm:mt-0">
+              {query?.uuid}
+            </dd>
+          </div>
+
+          <div className="px-4 py-2 sm:grid sm:grid-cols-3 sm:gap-4 sm:px-0">
+            <dt className="text-sm font-medium leading-6 text-gray-900">
+              Created At:
+            </dt>
+            <dd className="mt-1 text-sm leading-6 text-gray-700 sm:col-span-2 sm:mt-0">
+              {query?.createdAt}
+            </dd>
+          </div>
+
+          <div className="px-4 py-2 sm:grid sm:grid-cols-3 sm:gap-4 sm:px-0">
+            <dt className="text-sm font-medium leading-6 text-gray-900">
+              Type:
+            </dt>
+            <dd className="mt-1 text-sm leading-6 text-gray-700 sm:col-span-2 sm:mt-0">
+              {query?.type}
+            </dd>
+          </div>
+
+          <div className="px-4 py-2 sm:grid sm:grid-cols-3 sm:gap-4 sm:px-0">
+            <dt className="text-sm font-medium leading-6 text-gray-900">
+              Params:
+            </dt>
+            <dd className="mt-2 text-sm text-gray-900 sm:col-span-2 sm:mt-0">
+              <ul className="divide-y divide-gray-200 rounded-md border border-gray-200">
+                {Object.entries(
```

Comment:
again this seems to be quite involved and maybe require some logging in the future, is there a way to do it in JS and map to UI here

Reply:
good suggestion to move some of this out of the template. that said, the fun thing about react nowadays is you could add logging now a couple lines down (just before the return statement). I had one there as I was developing actually!